src/System.IO.FileSystem.AccessControl/src/System/IO/FileSystemAclExtensions.cs

```diff
@@ -10,6 +10,75 @@ namespace System.IO
 {
     public static class FileSystemAclExtensions
     {
+        /// <summary>Creates a new file and returns a <see cref="FileStream" /> instance representing it, ensuring the file is created with all the specified permissions and security properties.</summary>
+        /// <param name="fileInfo">The object describing a file that does not exist in disk yet.</param>
+        /// <param name="mode">A constant that determines how to open or create the file.</param>
+        /// <param name="rights">A constant that determines the access rights to use when creating access and audit rules.</param>
+        /// <param name="share">A constant that determines how the file will be shared by processes.</param>
+        /// <param name="bufferSize">A positive <see cref="int" /> value greater than 0 indicating the buffer size. The default buffer size is 4096.</param>
+        /// <param name="options">A constant that specifies additional file options.</param>
+        /// <param name="fileSecurity">An object that determines the access control and audit security for the file.</param>
+        /// <returns>A file stream that represents the newly created file.</returns>
+        /// <exception cref="ArgumentNullException"><paramref name="fileSecurity" /> is <see langword="null" />.</exception>
+        /// <exception cref="ArgumentOutOfRangeException"><paramref name="bufferSize" /> is negative or zero.
+        ///
+        /// -or-
+        ///
+        /// <paramref name="mode" />, <paramref name="rights" />, <paramref name="share" /> or <paramref name="options" /> contain an invalid value.
+        /// </exception>
+        /// <remarks><format type="text/markdown"><![CDATA[
+        /// ## Remarks
+        ///
+        /// This method can also throw any of the exceptions that <xref:System.IO.FileStream> can throw.
+        /// ]]></format></remarks>
+        public static FileStream Create(this FileInfo fileInfo, FileMode mode, FileSystemRights rights, FileShare share, int bufferSize, FileOptions options, FileSecurity fileSecurity)
+        {
+            string badArg = null;
```

Comment:
Based my checks on what we already do inside the FileStream constructor that receives all these arguments.
Question: Should I delegate the argument checks and exception throws to the other public APIs I will be calling, like the FileStream constructor, or should I also check them here?

Reply:
You want throwing to occur in the same order as much as possible, so I'm guessing you'll have to duplicate the checks.

Reply:
Ok thanks for confirming. I don't mind duplicating checks. I wanted to make sure I also had to throw here. I'll make sure they are thrown in the same order whenever possible.

Reply:
This will be addressed in the File Create PR.